improver/cli/apply_night_mask.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to mask night values within the input cube."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube):
+    """Masks night values from input cube to generate day only output.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be masked.
+
+    Returns:
+        iris.cube.Cube:
+	    Input cube with all night values masked out.
+
+    """
+
+    import numpy as np
+    import numpy.ma as ma
+
+    from improver.utilities.solar import DayNightMask
+
+    mask = DayNightMask()(cube).data
+    # Broadcast mask to shape of input cube to account for additional dimensions.
+    mask = np.broadcast_to(mask, cube.shape)
+    # masking night values from the data.
+    cube.data = ma.masked_where(mask == DayNightMask().night, cube.data)
```

Comment:
After thinking about it, I think that what we really want is for points that are in the night to return a UV index of zero. This CLI will currently return a cube where the night values are masked. Therefore I think that we actually want:
```suggestion
    cube.data = ma.where(mask == DayNightMask().night, 0, cube.data)
```
The comment above this line will need to be updated to be something like "set night values to be zero."

Reply:
done